### PR TITLE
Address CI issues

### PR DESCRIPTION
**Pull request by addyess**
## Overview
Address CI issues and appropriately build with charmcraft

## Details
* Updates the inclusive language check
* Updates the ci to run on python 3.8, 3.10, and 3.12

### CODE REVIEW

File: src/metadata.yaml [hunk 1]
```diff
@@ -1,27 +1,28 @@
...
 description: |
   keepalived is used for monitoring real servers within a Linux Virtual
   Server (LVS) cluster.
 docs: https://discourse.charmhub.io/t/keepalived-docs-index/6043
 tags:
-  - "networking"
-series:
-  - noble
-  - jammy
-  - focal
```
Comment (addyess) [inline src/metadata.yaml]: Removal of the `series` list (replaced by the charmcraft yaml bases list must be dropped to use the `assumes` metadata)

File: src/metadata.yaml [hunk 2]
```diff
@@ -1,27 +1,28 @@
...
+- networking
+
+assumes:
+- juju >= 3.4
```
Comment (addyess) [inline src/metadata.yaml]: Required b/c the addition of noble forces the charm to be used on juju 3.4 and beyond.  No prior juju client will support the inclusion of noble as a potential base

File: .github/workflows/tox.yaml [hunk 3]
```diff
@@ -4,7 +4,7 @@ on: [pull_request]
 jobs:
```
Comment (louiseschmidtgen) [inline .github/workflows/tox.yaml:9]: Nice to have: restrict the gh runner top-level permissions and add job level permissions as needed for the workflow like we do it in k8s-snap: https://github.com/canonical/k8s-snap/blob/main/.github/workflows/integration.yaml#L14

File: .github/workflows/tox.yaml [hunk 4]
```diff
@@ -22,17 +24,36 @@ jobs:
...
+        uses: actions/checkout@v4
       - name: Setup operator environment
         uses: charmed-kubernetes/actions-operator@main
         with:
           provider: vsphere
+          juju-channel: 3/stable
           credentials-yaml: ${{ secrets.CREDENTIALS_YAML }}
           clouds-yaml: ${{ secrets.CLOUDS_YAML }}
-          bootstrap-options: "--model-default datastore=vsanDatastore --model-default primary-network=VLAN_2764"
+          bootstrap-constraints: "arch=amd64 cores=2 mem=4G"
```
Comment (louiseschmidtgen) [inline .github/workflows/tox.yaml:40]: Sounds reasonable, where are these constraints coming from?

File: src/metadata.yaml [hunk 5]
```diff
@@ -1,27 +1,25 @@
...
-  - "Konstantinos Tsakalozos <kos.tsakalozos@canonical.com>"
-  - "Valentin Boucher <valentin.boucher@kontron.com>"
+- "Konstantinos Tsakalozos <kos.tsakalozos@canonical.com>"
+- "Valentin Boucher <valentin.boucher@kontron.com>"
 description: |
   keepalived is used for monitoring real servers within a Linux Virtual
   Server (LVS) cluster.
 docs: https://discourse.charmhub.io/t/keepalived-docs-index/6043
 tags:
-  - "networking"
```
Comment (louiseschmidtgen) [inline src/metadata.yaml:11]: Why are we removing the tag?

File: tests/data/bundle.yaml [hunk 6]
```diff
@@ -12,21 +12,9 @@ applications:
...
-- - kubeapi-load-balancer:certificates
-  - easyrsa:client
+- [kubernetes-control-plane:loadbalancer-external, kubeapi-load-balancer:lb-consumers]
+- [kubernetes-control-plane:loadbalancer-internal, kubeapi-load-balancer:lb-consumers]
+- [kubeapi-load-balancer:certificates,             easyrsa:client]
 #######################################
 #  keepalived relations
-- - keepalived:juju-info
-  - kubeapi-load-balancer:juju-info
-- - keepalived:lb-sink
```
Comment (louiseschmidtgen) [inline tests/data/bundle.yaml:27]: Why are these relations getting removed?

**Review by louiseschmidtgen — COMMENTED** [review]
Nice work Adam, I've got a few questions then it should be good to go.

File: .github/workflows/tox.yaml (hunk 4 above)
Comment (addyess) [reply .github/workflows/tox.yaml:40]: it's a common trait of other charmed-kubernetes charm workflows.  Much of these changes are upgrades to this workflow based on something like 
https://github.com/charmed-kubernetes/charm-kubernetes-control-plane/blob/main/.github/workflows/vsphere-integration.yaml#L19-L24

File: tests/data/bundle.yaml (hunk 6 above)
Comment (addyess) [reply tests/data/bundle.yaml:27]: well, mostly b/c they are unnecessary of the test, and secondarily -- charms like kubernetes-control-plane and kubernetes-worker no longer suppose these endpoints.

File: .github/workflows/tox.yaml (hunk 3 above)
Comment (addyess) [reply .github/workflows/tox.yaml:9]: Done!  nice

**Comment by addyess** [comment]
Waiting on `charms.reactive` to be published

**Comment by addyess** [comment]
> Waiting on `charms.reactive` to be published

Published now!!
https://pypi.org/project/charms.reactive/

**Pull request merged**